trieste/acquisition/function.py

```diff
@@ -1287,7 +1287,8 @@ def gibbon(
     except NotImplementedError:
```

Comment:
so only homoscedastic GP models will have `get_observation_noise` method? should the model be something more specific than `ProbabilisticModel`?

Reply:
I need to think about this, as quite a few of our acq functions are not really defined properly for *any* prob model. This will be another PR though

Reply:
hm, true, more of a design question, how to do these checks in general - here I'm just wondering whether checking for existence of this method is the best way to ensure its a homoscedastic GP, perhaps its best left as an assumption that is stressed in the docstring and leave it at that

also, scratch my comment about more specific than `ProbabilisticModel` class, I'm supposedly trying to avoid importing gpflow-related stuff elsewhere

Reply:
above it should be an `assert` statement rather than if-else:
```
if len(samples) == 0:
        raise ValueError("Min-value samples must be populated.")
```

Reply:
scratch that, after a discussion on recent Joel's PR, we should probably use `tf.debugging` assertions everywhere